### PR TITLE
Forager skill for companions

### LINKED ISSUE #78: Forager personal skill
Create support for forager personal skill. Add a forager controller.

### PR DESCRIPTION
**Pull request by purrplingcat**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Content updates (new dialogues added or some string edited and etc)
- [ ] **Merge ASAP**
- [ ] Merge before *enumerated issues or other PRs numbers*
- [ ] Merge after *enumerated issues or other PRs numbers*

## Description
Adds a forager skill for companion, the forager controller and refactor of some code around.

Foraging works in this way:

- When companion can forage, do forage walk to a point
- If on point is a bush, shake with it
- If point is an empty tile, grab here (with animation)
- If forager found something, do emote `!` or `<3`
- When is foraging action done, companion returns to player
- If player is so so far from forager, then companion react with say a "waitonme" dialogue and friendship points decrecreases (-5 points). Forager can change share forages mind immediatelly and don't share last foraged item with player.
- Foraged items can share with player
- Forager can change a mind and found item don't share with player
- If forager something found, there is a base chance 50% they will share it with player. Mor friendship hearts means more chance (>50%) the found item to be shared. The base for 50% is a 5 hearts. If you have under 5 hearts (and allowed recruitment under this heart level), then the share chance is < 50%.
- In indoors don't forage

### Specialities

- Forage a tree gives a tree forage (like maple seed, acorn and pine) if tree is grown.
- Forage a maple in fall gives a hazelnut.
- Forage a bush gives a salmonberry (spring) or blackbarry (fall) when bush is in bloom
- Forage a bush rarely gives a rare/special forage item (chance is <1%)

### NPCs assigned to forager skill

- Leah

## How to test
Recruit a forager and test all described features above. Test it in all known locations.

## Checklist

**Please check if the PR fulfills these requirements**
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
  - [ ] All new dialogue keys was introduces in dialogue docs
- [ ] Changes was tested and passed

## Other information
---

Closes #78 
Refs #2

### CODE REVIEW

**Comment by purrplingcat** [comment]
Maybe Emily will be assigned as forager.

File: NpcAdventure/NpcAdventureMod.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using NpcAdventure.Compatibility;
 using NpcAdventure.Story;
 using NpcAdventure.Story.Scenario;
+using StardewValley;
```
Comment (purrplingcat) [inline NpcAdventure/NpcAdventureMod.cs]: Unused import, remove it.

File: NpcAdventure/AI/AI_StateMachine.cs [hunk 2]
```diff
@@ -20,13 +20,14 @@ namespace NpcAdventure.AI
     /// <summary>
     /// State machine for companion AI
     /// </summary>
-    internal partial class AI_StateMachine : Internal.IUpdateable, Internal.IDrawable
+    internal partial class  AI_StateMachine : Internal.IUpdateable, Internal.IDrawable
```
Comment (CallMeFoxie) [inline NpcAdventure/AI/AI_StateMachine.cs:23]: extra space?

**Comment by purrplingcat** [comment]
> too much code with zero comments. If it works, ship it.

@CallMeFoxie Should I add more comments and summary docs?

**Comment by CallMeFoxie** [comment]
> > too much code with zero comments. If it works, ship it.
> 
> @CallMeFoxie Should I add more comments and summary docs?

there's not that many comments to begin with so.. no need to probably ;)

**Pull request merged**